app/js/redux-store.jsx

```diff
@@ -11,23 +11,28 @@ import {
   createStore, applyMiddleware, combineReducers, compose,
 } from 'redux';
 import createSagaMiddleware from 'redux-saga';
+import promiseMiddleware from 'redux-promise-middleware';
+import { createLogger } from 'redux-logger';
 import { sagas as openmrsSagas, reducers as openmrsReducers } from '@openmrs/react-components';
-import { reducer as reduxFormReducer } from 'redux-form';
-
-import * as reducers from './reducers';
+// import reducers from './reducers';
+import reducers from './reducers';
 
 const sagaMiddleware = createSagaMiddleware();
 
-const middlewares = [sagaMiddleware];
+const promiseTypeSuffixes = ['LOADING', 'SUCCESS', 'FAILURE'];
+
+const middlewares = [
+  sagaMiddleware,
+  promiseMiddleware({ promiseTypeSuffixes }),
```

Comment:
I had to remind myself how it worked again, but I do think I really like this promise middleware you introduced us to... :)

Reply:
I would have to write an errorMiddleWare to handle errors in failed server responses though but since we're not making any ajax calls just yet, it's good atm